exporter/otlphttpexporter/otlp_test.go

```diff
@@ -467,22 +467,58 @@ func TestPartialResponse_missingHeaderButHasBody(t *testing.T) {
 	exp, err := newExporter(cfg, set)
 	require.NoError(t, err)
 
-	response := ptraceotlp.NewExportResponse()
-	partial := response.PartialSuccess()
-	partial.SetErrorMessage("hello")
-	partial.SetRejectedSpans(1)
-	data, err := response.MarshalProto()
-	require.NoError(t, err)
-	resp := &http.Response{
-		// `-1` indicates a missing Content-Length header in the Go http standard library
-		ContentLength: -1,
-		Body:          io.NopCloser(bytes.NewReader(data)),
-		Header: map[string][]string{
-			"Content-Type": {"application/x-protobuf"},
-		},
+	contentTypes := []struct {
+		contentType string
+	}{
+		{contentType: "application/x-protobuf"},
+		{contentType: "application/json"},
+	}
+	for _, tt := range contentTypes {
+		for _, telemetryType := range []string{"logs", "metrics", "traces"} {
+			t.Run("Missing header but has body for content type "+tt.contentType+" "+telemetryType, func(t *testing.T) {
+				response := ptraceotlp.NewExportResponse()
```

Comment:
For success they probably all look the same, but technically each signal has their own expected response object.  Can you update the test to use the proper response per signal?

Reply:
good catch, will update.

Reply:
I _think_ I now have the tests in a good place. I apologize for so much back and forth as I am getting used to the project details.